app/controllers/actions/CheckForSubmissionAction.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import controllers.routes
+import models.requests.DataRequest
+import play.api.libs.json.Json
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{ActionRefiner, Result}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckForSubmissionActionImpl @Inject() (implicit val executionContext: ExecutionContext) extends CheckForSubmissionAction {
+
+  override protected def refine[A](request: DataRequest[A]): Future[Either[Result, DataRequest[A]]] =
+    if (request.userAnswers.data == Json.obj()) {
```

Comment:
Where is this used?
Looks like it checks for an empty useranswers and redirects

Reply:
It's used in `CheckYourAnswersController.onPageLoad()` to check if the form has already been submitted (the user answers will get emptied after submission but that's within the scope of another ticket).